R/rdeephaven/R/client_wrapper.R

```diff
@@ -0,0 +1,102 @@
+                stop("'credentials' must be passed as a list of two strings representing a key/value pair for authentication.")
+            }
+
+            # now value checking
+            if (!(sessionType == "python" || sessionType == "groovy")) {
+                warning("The session type you provided is not currently supported. Defaulting to 'python'.")
+            }
+            if (!(authType == "default" || authType == "basic" || authType == "custom")) {
+                warning("The authentication type you provided is not currently supported. Defaulting to 'default'.")
+            }
+
+            private$internal_client <- new(INTERNAL_Client, target=target, sessionType=sessionType,
+                                           authType=authType, key=credentials[[1]], value=credentials[[2]])
+        },
+
+        # retrieve a TableHandle reference to a named table on the server
+        open_table = function(tableName) {
+            private$verify_string(tableName, "tableName")
+            if (!private$check_for_table(tableName)) {
+                stop("The table you're trying to pull does not exist on the server.")
+            }
+            this_dh_table_handle = private$internal_client$open_table(tableName)
+            return(NewTableHandle$new(this_dh_table_handle))
+        },
+
+        # import an R object to a table to the server, and retrieve a TableHandle reference to that table
+        import_table = function(tableObject) {
+            if (!(class(tableObject)[[1]] == "data.frame" ||
+                  class(tableObject)[[1]] == "RecordBatchReader" ||
+                  (length(class(tableObject)) == 4 &&
+                   class(tableObject)[[1]] == "Table" &&
+                   class(tableObject)[[3]] == "ArrowObject"))) {
+                stop("'tableObject' must be either an R Data Frame, an Arrow Table, or an Arrow Record Batch Reader.")
+            }
+            if (class(tableObject)[[1]] == "data.frame") {
+                this_dh_table_handle = private$df_to_dh_table(tableObject)
+            }
+            else if (class(tableObject)[[1]] == "RecordBatchReader") {
+                this_dh_table_handle = private$rbr_to_dh_table(tableObject)
+            }
+            else if ((length(class(tableObject)) == 4 &&
+                      class(tableObject)[[1]] == "Table" &&
+                      class(tableObject)[[3]] == "ArrowObject")) {
+                this_dh_table_handle = private$arrow_to_dh_table(tableObject)
+            }
+            else {
+                stop("An unexpected type was passed to 'push' and made it through our stop handling. 'tableObject' must be either an R Data Frame, an Arrow Table, or an Arrow Record Batch Reader.")
+            }
+            return(NewTableHandle$new(this_dh_table_handle))
+        },
+
+        run_script = function(script) {
+            private$verify_string(script, "script")
+            private$internal_client$run_script(script)
```

Comment:
This has me wondering about c++ error handling.  What happens if C++ throws an exception?  Do we behave ok?